tests/integ/test_transformer.py

```diff
@@ -301,6 +302,54 @@ def test_transform_byo_estimator(sagemaker_session):
         assert tags == model_tags
 
 
+def test_stop_transform_job(sagemaker_session, mxnet_full_version):
+    data_path = os.path.join(DATA_DIR, "mxnet_mnist")
+    script_path = os.path.join(data_path, "mnist.py")
+    tags = [{"Key": "some-tag", "Value": "value-for-tag"}]
+
+    mx = MXNet(
+        entry_point=script_path,
+        role="SageMakerRole",
+        train_instance_count=1,
+        train_instance_type="ml.c4.xlarge",
+        sagemaker_session=sagemaker_session,
+        framework_version=mxnet_full_version,
+    )
+
+    train_input = mx.sagemaker_session.upload_data(
+        path=os.path.join(data_path, "train"), key_prefix="integ-test-data/mxnet_mnist/train"
+    )
+    test_input = mx.sagemaker_session.upload_data(
+        path=os.path.join(data_path, "test"), key_prefix="integ-test-data/mxnet_mnist/test"
+    )
+    job_name = unique_name_from_base("test-mxnet-transform")
+
+    with timeout(minutes=TRAINING_DEFAULT_TIMEOUT_MINUTES):
+        mx.fit({"train": train_input, "test": test_input}, job_name=job_name)
+
+    transform_input_path = os.path.join(data_path, "transform", "data.csv")
+    transform_input_key_prefix = "integ-test-data/mxnet_mnist/transform"
+    transform_input = mx.sagemaker_session.upload_data(
+        path=transform_input_path, key_prefix=transform_input_key_prefix
+    )
+
+    transformer = mx.transformer(1, "ml.m4.xlarge", tags=tags)
+    transformer.transform(transform_input, content_type="text/csv")
+
+    time.sleep(15)
+
+    latest_transform_job_name = transformer.latest_transform_job.name
+
+    print("Attempting to stop {}".format(latest_transform_job_name))
+
+    transformer.stop_transform_job()
+
+    desc = transformer.latest_transform_job.sagemaker_session.sagemaker_client.describe_transform_job(
+        TransformJobName=latest_transform_job_name
+    )
+    assert desc["TransformJobStatus"] == "Stopping"
```

Comment:
Is there a way to check if the transform job stopping is complete?

Reply:
Hey @ChuyangDeng We can add a wait parameter to the Transformer.stop_transform_job() method....waiting for the stopping to complete. 
If the concern is only with the test case, after the assertion, we can add the following lines:
    
    transformer.wait()
    desc = transformer.latest_transform_job.sagemaker_session.sagemaker_client.describe_transform_job(
        TransformJobName=latest_transform_job_name
    )
    assert desc["TransformJobStatus"] == "Stopped"

Please let me know your thoughts on this.

Reply:
I think it'd be better to have the stop method block on the job successfully stopping (not just for the test case)